addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/handler/AirVisualNodeHandler.java

```diff
@@ -0,0 +1,271 @@
+        }
+        this.nodePassword = config.password;
+
+        this.nodeShareName = config.share;
+
+        this.refreshInterval = config.refresh * 1000L;
+
+        schedulePoll();
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Node poll scheduled");
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), true);
+        } else {
+            logger.debug("Can not handle command '{}'", command);
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(false);
+        }
+    }
+
+    private synchronized void schedulePoll() {
+        stopPoll();
+        logger.debug("Scheduling poll for 500ms out, then every {} ms", refreshInterval);
+        pollFuture = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                poll();
+            }
+        }, 500, refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollNode();
+            updateStatus(ThingStatus.ONLINE);
+        } catch (IOException e) {
+            logger.debug("Could not connect to Node", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (Exception e) {
+            logger.error("Unexpected error connecting to Node", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void pollNode() throws IOException {
+        String jsonData = getNodeJsonData();
+        NodeData currentNodeData = gson.fromJson(jsonData, NodeData.class);
+        if (nodeData == null || currentNodeData.getStatus().getDatetime() > nodeData.getStatus().getDatetime()) {
+            nodeData = currentNodeData;
+            // Update all channels from the updated Node data
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId(), false);
+            }
+        }
+    }
+
+    private String getNodeJsonData() throws IOException {
+        String url = "smb://" + nodeAddress +"/" + nodeShareName + "/" + NODE_JSON_FILE;
+        NtlmPasswordAuthentication auth = new NtlmPasswordAuthentication(null, nodeUsername, nodePassword);
+        try (SmbFileInputStream in = new SmbFileInputStream(new SmbFile(url, auth))) {
+            return IOUtils.toString(new SmbFileInputStream(new SmbFile(url, auth)), StandardCharsets.UTF_8.name());
```

Comment:
You probably ment to use the variable `in` here that you moved to the line above.